trace_context/HTTP_HEADER_FORMAT.md

```diff
@@ -0,0 +1,159 @@
+
+#### Trace-id
+
+Is the ID of the whole trace forest. It is represented as a 16-bytes array, e.g., 
+`4bf92f3577b34da6a3ce929d0e0e4736`. All bytes 0 is considered invalid.
+
+Implementation may decide to completely ignore the traceparent if the trace-id is invalid.
+
+#### Span-id
+
+Is the ID of the caller span (parent). It is represented as a 8-bytes array, e.g., 
+`00f067aa0ba902b7`. All bytes 0 is considered invalid.
+
+Implementation may decide to completely ignore the traceparent if the span-id is invalid.
+
+#### Trace-options
+
+Controls tracing options such as sampling, trace level etc. It is a 1-byte representing a 8-bit 
+unsigned integer. The flags are recommendations given by the caller rather than strict rules to 
+follow for 3 reasons:
+
+1. Trust and abuse.
+2. Bug in caller
+3. Different load between caller service and callee service might force callee to down sample.    
+
+##### Bits behavior definition (01234567):
+* The least significant bit (the 7th bit) provides recommendation whether the request should be 
+traced or not (`1` recommends the request should be traced, `0` means the caller does not
+make a decision to trace and the decision might be deferred). When `trace-options` is missing
+the default value for this bit is `0`
+* The behavior of other bits is currently undefined.
+
+#### Examples of HTTP headers
+
+*Valid sampled traceparent:*
+
+```
+Value = 00-4bf92f3577b34da6a3ce929d0e0e4736-00f067aa0ba902b7-01
+base16(<Version>) = 00
+base16(<TraceId>) = 4bf92f3577b34da6a3ce929d0e0e4736
+base16(<SpanId>) = 00f067aa0ba902b7
+base16(<TraceOptions>) = 01  // sampled
+```
+
+*Valid not-sampled traceparent:*
+
+```
+Value = 00-4bf92f3577b34da6a3ce929d0e0e4736-00f067aa0ba902b7-00
+base16(<Version>) = 00
+base16(<TraceId>) = 4bf92f3577b34da6a3ce929d0e0e4736
+base16(<SpanId>) = 00f067aa0ba902b7
+base16(<TraceOptions>) = 00  // not-sampled
+```
+
+# Trace-State Field
+
+## Header name
+
+`tracestate`
+
+As multiple trace graphs can be present, multiple trace state headers are
+allowed. Values can be combined in a single header according to the [RFC 7230](https://tools.ietf.org/html/rfc7230#page-24).
+
+## Header value
+
+`name1[=value1[;properties1]],name2[=value2[;properties2]]`
+
+**Limits:**
+Maximum length of a combined header MUST be less than 512 bytes. 
+
+## Name format
+
+Url encoded string. Spaces are allows before and after the name. Header with the trimmed name and with spaces before and after name MUST be considered identical.
+
+Names `id`, `span-id`, `trace-id`, `sampled` are reserved. These properties are defined in `traceparent` header.
+
+## Value format
+
+Value starts after equal sign and ends with the special character `;`, separator `,` or end of string. Value represents a url encoded string and case sensitive. Spaces are allowed in the beginning and the end of the value. Value with spaces before and after MUST be considered identical to the trimmed value. 
+
+## Properties
```

Comment:
Based on other comments and my own personal opinion, I'd favor chopping this entire `Properties` section and just call out that the vendor value is opaque and vendors can do whatever they want as long as they don't use value-ending characters (e.g. `,`) or otherwise violate the RFC 7230 spec.

I'd also be perfectly happy with @adriancole's suggestion that maybe the vendor value should be explicitly base64 rather than free-for-all. I might even prefer that - it means some extra work when parsing/extracting your trace graph, but it could seriously reduce confusion and misinterpretation when humans visually look at a `tracestate` header and try to understand it without reading this specification.

Reply:
And that confusion/misinterpretation is inevitable when `key=val` pairs are embedded everywhere in a large header value.

Reply:
done